### ISSUE TITLE
Cache cleanup does not clean cache entries for versions of gradle >8.9

### ISSUE THREAD

**Issue by benkay** [gradle/actions#417]
The [documentation](https://github.com/gradle/actions/blob/main/docs/setup-gradle.md#configuring-cache-cleanup) implies that no further setup is required for the action to cleanup caches after builds.

I was investigating why our CI cache is steadily rising, and the cause seems to be that old transforms are not being cleaned up effectively.

We're building using Gradle 8.10.2 (via the wrapper), but this is the log output from the cache cleanup phase of the action

```
   Extracted Gradle 8.9 to /opt/actions-runner/_work/_temp/.gradle-actions/gradle-installations/installs/gradle-8.9
  Provisioned Gradle executable /opt/actions-runner/_work/_temp/.gradle-actions/gradle-installations/installs/gradle-8.9/bin/gradle
[..snip..]
  artifact cache (/home/runners/.gradle/caches/modules-2) cleaned up in 0.121 secs.
  Artifact transforms cache (/home/runners/.gradle/caches/8.9/transforms) removing files not accessed on or after Fri Oct 18 13:13:24 UTC 2024.
  Artifact transforms cache (/home/runners/.gradle/caches/8.9/transforms) cleanup deleted 0 files/directories.
  Artifact transforms cache (/home/runners/.gradle/caches/8.9/transforms) cleaned up in 0.0 secs.
  groovy-dsl (/home/runners/.gradle/caches/8.9/groovy-dsl) removing files not accessed on or after Fri Oct 18 13:13:24 UTC 2024.
  groovy-dsl (/home/runners/.gradle/caches/8.9/groovy-dsl) cleanup deleted 2 files/directories.
  groovy-dsl (/home/runners/.gradle/caches/8.9/groovy-dsl) cleaned up in 0.009 secs.
  jars (/home/runners/.gradle/caches/jars-9) cleanup deleted 0 files/directories.
  jars (/home/runners/.gradle/caches/jars-9) removing files not accessed on or after Fri Oct 18 13:13:24 UTC 2024.
  jars (/home/runners/.gradle/caches/jars-9) cleanup deleted 1 files/directories.
  jars (/home/runners/.gradle/caches/jars-9) cleaned up in 0.039 secs.
```

As the cache cleanup is provisioning Gradle 8.9, it looks like it's not cleaning up any transforms produced by newer versions.
The cache cleanup should either use the wrapper, or provision the latest version of Gradle so that transform caches are cleaned up.

**Comment by bigdaz** [comment]
Thanks for the report. This is likely an unwanted side-effect of #364, which performs cache-cleanup using the version of Gradle on PATH (at least 8.9) instead of always downloading the latest.

This means that cache cleanup won't work for a Gradle version > 8.9 unless that version (or newer) is available on the PATH.

**Comment by Su5eD** [comment]
Running into the same issue here. I tried manually specifying the used gradle version but it looks like post-cleanup provisions its own gradle instance?

Is there an temporary workaround I can use until this is addressed?

**Comment by TemMax** [comment]
Thanks, @ViacheslavKudinov . The workaround is working like a charm!

**Comment by TemMax** [comment]
Changing the flag from `true` to `false` worked for me: `gradle-home-cache-cleanup: false`

**Comment by bigdaz** [comment]
Thanks for the additional reports. 
Yesterday I raised #431 for the specific issue of cache-cleanup failing on Gradle 8.11.
The fix has been merged and will be included in the next `gradle/actions` release, likely today.

**Comment by bigdaz** [comment]
The recent release of `setup-gradle@v4.2.0` has addressed 2 issues related to cache cleanup:
- #431: Cache cleanup is fixed for Gradle 8.11.  This fixes the issue that reports `Cannot get the value of write-only property 'removeUnusedEntriesOlderThan'`. 
- #417 (this issue). Cache cleanup now runs with Gradle 8.11, which means that entries written by any currently released version of Gradle will be cleaned up.

There is still an issue with cache-cleanup failing to remove entries written by newer (not-yet-released) versions of Gradle. I've raised #436 for this issue.
